lib/js/asyncjs.nim

```diff
@@ -176,44 +187,70 @@ when defined(nimExperimentalAsyncjsThen):
 
     type OnReject* = proc(reason: Error)
 
-    proc then*[T, T2](future: Future[T], onSuccess: proc(value: T): T2, onReject: OnReject = nil): Future[T2] =
+    proc then*[T, Fun](future: Future[T], onSuccess: Fun, onReject: OnReject = nil): auto =
```

Comment:
how about T2: proc
does this work?

Reply:
fixed, thanks! i had tried `then*[T, Fun: proc]` which didn't work, but this works: `then*[T; Fun: proc]`

I think `,` vs `;` is not documented for generic params, we should document it

Reply:
Could you use Func or something else instead of Fun which looks a bit strange to me :)

Reply:
`Func` is bad because it would suggest only `func` is accepted. But I've found a better way:
```nim
proc then*[T](future: Future[T], onSuccess: proc, onReject: OnReject = nil): auto =
```
in other similar caes, what name would you have suggested?

Reply:
ok

Reply:
=> PR https://github.com/nim-lang/Nim/pull/17192